diff --git a/votingapi.module b/votingapi.module
index 983b5dd..e76a7c1 100644
--- a/votingapi.module
+++ b/votingapi.module
@@ -314,13 +314,18 @@ function votingapi_delete_results($vote_results = array()) {
  *   An array of votes matching the criteria.
  */
 function votingapi_select_votes($criteria = array(), $limit = 0) {
-  $anon_window = variable_get('votingapi_anonymous_window', 3600);
+  $anon_window = variable_get('votingapi_anonymous_window', 86400);
   if (!empty($criteria['vote_source']) && $anon_window >= 0) {
     $criteria['timestamp'] = REQUEST_TIME - $anon_window;
   }
   $query = db_select('votingapi_vote')->fields('votingapi_vote');
   foreach ($criteria as $key => $value) {
-    $query->condition($key, $value, is_array($value) ? 'IN' : '=');
+    if ($key == 'timestamp') {
+      $query->condition($key, $value, '>');
+    }
+    else {
+      $query->condition($key, $value, is_array($value) ? 'IN' : '=');
+    }
   }
   if (!empty($limit)) {
     $query->range(0, $limit);
