diff --git xmlsitemap_menu/xmlsitemap_menu.module xmlsitemap_menu/xmlsitemap_menu.module
index 7200b32..6eba673 100644
--- xmlsitemap_menu/xmlsitemap_menu.module
+++ xmlsitemap_menu/xmlsitemap_menu.module
@@ -84,7 +84,7 @@ function xmlsitemap_menu_xmlsitemap_process_menu_links(array $mlids, array $xmls
  * @see xmlsitemap_add_link_bundle_settings()
  */
 function xmlsitemap_menu_form_menu_edit_menu_alter(&$form, $form_state) {
-  $menu = isset($form['menu_name']['#default_value']) ? $form['menu_name']['#default_value'] : '';
+  $menu = isset($form['menu_name']['#value']) ? $form['menu_name']['#value'] : '';
   module_load_include('inc', 'xmlsitemap', 'xmlsitemap.admin');
   xmlsitemap_add_link_bundle_settings($form, $form_state, 'menu', $menu);
 }
@@ -153,28 +153,10 @@ function xmlsitemap_menu_menu_link_insert(array $link) {
 
 /**
  * Implements hook_menu_link_update().
- *
- * @see hook_menu_link_alter()
  */
 function xmlsitemap_menu_menu_link_update(array $link) {
-  //$link += array('xmlsitemap' => array());
-  //xmlsitemap_menu_xmlsitemap_process_menu_links(array($link['mlid']), $link['xmlsitemap']);
-}
-
-/**
- * Implements hook_menu_link_alter().
- *
- * We have to use this hook rather than hook_menu_link_update() because this
- * hook is not always called if the user does not edit the core menu item
- * fields.
- *
- * @see http://drupal.org/node/1013856
- */
-function xmlsitemap_menu_menu_link_alter(array &$link) {
-  if (!empty($link['mlid'])) {
-    $link += array('xmlsitemap' => array());
-    xmlsitemap_menu_xmlsitemap_process_menu_links(array($link['mlid']), $link['xmlsitemap']);
-  }
+  $link += array('xmlsitemap' => array());
+  xmlsitemap_menu_xmlsitemap_process_menu_links(array($link['mlid']), $link['xmlsitemap']);
 }
 
 /**
