diff --git a/date_all_day/date_all_day.module b/date_all_day/date_all_day.module
index 8e30e44..c452521 100644
--- a/date_all_day/date_all_day.module
+++ b/date_all_day/date_all_day.module
@@ -198,7 +198,7 @@ function date_all_day_date_combo_process_alter(&$element, &$form_state, $context
     $date_is_all_day = date_is_all_day($from, $to);
     $all_day = !empty($form_state['values']['all_day']) || $date_is_all_day;
     $element['all_day'] = array(
-      '#title' => t('All Day'),
+      '#title' => t('All day'),
       '#type' => 'checkbox',
       '#default_value' => $all_day,
       '#weight' => -21,
@@ -417,4 +417,4 @@ function date_all_day_date_field_widget_settings_form_alter(&$form, $context) {
     $form['display_all_day']['#type'] = 'hidden';
     $form['display_all_day']['#value'] = 0;
   }
-}
\ No newline at end of file
+}
diff --git a/date_api/date.css b/date_api/date.css
index 4fdbc08..55e2b2e 100644
--- a/date_api/date.css
+++ b/date_api/date.css
@@ -227,3 +227,10 @@ div.date-calendar-day span.year {
 .date-year-range-select {
   margin-right: 1em;
 }
+
+.field-widget-date-select .form-item.form-type-date-select > label {
+  margin-top: 0.5em;
+  padding-left: 10px;
+  margin-bottom: -10px;
+  text-transform: uppercase;
+}
diff --git a/date_elements.inc b/date_elements.inc
index 21a4057..69db719 100644
--- a/date_elements.inc
+++ b/date_elements.inc
@@ -273,7 +273,7 @@ function date_combo_element_process($element, &$form_state, $form) {
 
   $show_todate = !empty($form_state['values']['show_todate']) || !empty($element['#default_value'][$to_field]);
   $element['show_todate'] = array(
-    '#title' => t('Show End Date'),
+    '#title' => t('Enter different end date'),
     '#type' => 'checkbox',
     '#default_value' => $show_todate,
     '#weight' => -20,
@@ -330,9 +330,9 @@ function date_combo_element_process($element, &$form_state, $form) {
   // is the 'Start' and which is the 'End' .
 
   if (!empty($field['settings']['todate'])) {
-    // Add class to allow date parts to float together on the same line.
-    $element[$from_field]['#title']  = ''; // The title is in the fieldset.
+    $element[$from_field]['#title'] = t('Start date');
     $element[$to_field] = $element[$from_field];
+    $element[$to_field]['#title'] = t('End date');
     $element[$to_field]['#default_value'] = isset($element['#default_value'][$to_field]) ? $element['#default_value'][$to_field] : '';
     $element[$to_field]['#required'] = FALSE;
     $element[$to_field]['#weight'] += .2;
@@ -619,4 +619,4 @@ function date_input_format($element, $field, $instance) {
     return $instance['widget']['settings']['input_format'];
   }
   return variable_get('date_format_short', 'm/d/Y - H:i');
-}
\ No newline at end of file
+}
