diff --git a/remote_stream_wrapper.inc b/remote_stream_wrapper.inc
index 492272b..7631521 100644
--- a/remote_stream_wrapper.inc
+++ b/remote_stream_wrapper.inc
@@ -61,6 +61,10 @@ class DrupalRemoteStreamWrapper implements DrupalStreamWrapperInterface {
    * Base implementation of getMimeType().
    */
   public static function getMimeType($uri, $mapping = NULL) {
+    if (parse_url($uri, PHP_URL_HOST) === parse_url($GLOBALS['base_url'], PHP_URL_HOST)) {
+      return DrupalLocalStreamWrapper::getMimeType($uri, $mapping);
+    }
+
     if (!isset($mapping)) {
       // The default file map, defined in file.mimetypes.inc is quite big.
       // We only load it when necessary.
