diff --git a/includes/media.fields.inc b/includes/media.fields.inc
index 74c5790..e2fa457 100644
--- a/includes/media.fields.inc
+++ b/includes/media.fields.inc
@@ -17,7 +17,7 @@ function media_field_info() {
         'file_extensions' => media_variable_get('file_extensions'),
       ),
       'default_widget' => 'media_generic',
-      'default_formatter' => 'media_large',
+      'default_formatter' => 'media',
       'property_type' => 'field_item_file',
       'property_callbacks' => array('entity_metadata_field_file_callback'),
     ),