diff --git a/token.css b/token.css
index 5c3b3ba..8596ce5 100644
--- a/token.css
+++ b/token.css
@@ -9,6 +9,10 @@
   padding-bottom: 0;
 }
 
+.token-tree tr .token-name {
+  padding-left: 25px;
+}
+
 .token-group {
   font-weight: bold;
 }
diff --git a/token.pages.inc b/token.pages.inc
index cf4c2d1..74ede48 100644
--- a/token.pages.inc
+++ b/token.pages.inc
@@ -132,7 +132,8 @@ function _token_token_tree_format_row($token, array $token_info, $is_group = FAL
 
   $row = $defaults;
   $row['id'] = _token_clean_css_identifier($token);
-  $row['data']['name'] = $token_info['name'];
+  $row['data']['name']['data'] = $token_info['name'];
+  $row['data']['name']['class'] = array('token-name');
   $row['data']['description'] = $token_info['description'];
 
   if ($is_group) {
