diff --git a/taxonomy_access.install b/taxonomy_access.install
index a8dffc5..f01dff1 100644
--- a/taxonomy_access.install
+++ b/taxonomy_access.install
@@ -16,35 +16,41 @@ function taxonomy_access_last_removed() {
 /**
  * Implements hook_install().
  *
- * Adds tables to database: 'taxonomy_access_term', 'taxonomy_access_default'
+ * Adds default grants to database.
  */
 function taxonomy_access_install() {
-
+  $query = db_insert('taxonomy_access_default')
+    ->fields(array('vid', 'rid', 'grant_view', 'grant_update', 'grant_delete', 'grant_create', 'grant_list'));
   // Default global perms for roles 1 (anonymous) and 2 (authenticated).
-  db_query(
-    'INSERT INTO {taxonomy_access_default}
-    (vid, rid, grant_view, grant_update, grant_delete, grant_create, grant_list)
-    VALUES
-    (:vid, :rid, :node_allow, :ignore, :ignore, :term_allow, :term_allow)',
-    array(
-      ':vid' => TAXONOMY_ACCESS_GLOBAL_DEFAULT,
-      ':rid' => DRUPAL_ANONYMOUS_RID,
-      ':node_allow' => TAXONOMY_ACCESS_NODE_ALLOW,
-      ':ignore' => TAXONOMY_ACCESS_NODE_IGNORE,
-      ':term_allow' => TAXONOMY_ACCESS_TERM_ALLOW)
-  );
-  db_query(
-    'INSERT INTO {taxonomy_access_default}
-    (vid, rid, grant_view, grant_update, grant_delete, grant_create, grant_list)
-    VALUES
-    (:vid, :rid, :node_allow, :ignore, :ignore, :term_allow, :term_allow)',
-    array(
-      ':vid' => TAXONOMY_ACCESS_GLOBAL_DEFAULT,
-      ':rid' => DRUPAL_AUTHENTICATED_RID,
-      ':node_allow' => TAXONOMY_ACCESS_NODE_ALLOW,
-      ':ignore' => TAXONOMY_ACCESS_NODE_IGNORE,
-      ':term_allow' => TAXONOMY_ACCESS_TERM_ALLOW)
-  );
+  $query->values(array(
+    'vid' => TAXONOMY_ACCESS_GLOBAL_DEFAULT,
+    'rid' => DRUPAL_ANONYMOUS_RID,
+    'grant_view' => TAXONOMY_ACCESS_NODE_ALLOW,
+    'grant_update' => TAXONOMY_ACCESS_NODE_IGNORE,
+    'grant_delete' => TAXONOMY_ACCESS_NODE_IGNORE,
+    'grant_create' => TAXONOMY_ACCESS_TERM_ALLOW,
+    'grant_list' => TAXONOMY_ACCESS_TERM_ALLOW));
+  $query->values(array(
+    'vid' => TAXONOMY_ACCESS_GLOBAL_DEFAULT,
+    'rid' => DRUPAL_AUTHENTICATED_RID,
+    'grant_view' => TAXONOMY_ACCESS_NODE_ALLOW,
+    'grant_update' => TAXONOMY_ACCESS_NODE_IGNORE,
+    'grant_delete' => TAXONOMY_ACCESS_NODE_IGNORE,
+    'grant_create' => TAXONOMY_ACCESS_TERM_ALLOW,
+    'grant_list' => TAXONOMY_ACCESS_TERM_ALLOW));
+  // Default global perm for administrator role.
+  $admin_role = variable_get('user_admin_role', FALSE);
+  if ($admin_role) {
+    $query->values(array(
+      'vid' => TAXONOMY_ACCESS_GLOBAL_DEFAULT,
+      'rid' => $admin_role,
+      'grant_view' => TAXONOMY_ACCESS_NODE_ALLOW,
+      'grant_update' => TAXONOMY_ACCESS_NODE_ALLOW,
+      'grant_delete' => TAXONOMY_ACCESS_NODE_ALLOW,
+      'grant_create' => TAXONOMY_ACCESS_TERM_ALLOW,
+      'grant_list' => TAXONOMY_ACCESS_TERM_ALLOW));
+  }
+  $query->execute();
 }
 
 /**
