From 4ace333ab4d8abfcc7e7c9d58ccb1b3d565ea108 Mon Sep 17 00:00:00 2001
From: Katherine Bailey <katherine@katbailey.net>
Date: Tue, 10 Apr 2012 14:09:24 -0400
Subject: [PATCH] Fixing the logic in the microdata property info callback

---
 includes/fivestar.field.inc |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/includes/fivestar.field.inc b/includes/fivestar.field.inc
index b99f041..38f37ed 100644
--- a/includes/fivestar.field.inc
+++ b/includes/fivestar.field.inc
@@ -752,10 +752,9 @@ function _fivestar_get_microdata_property_info($entity_type, $entity, $field, $i
       $info[$entity_type]['bundles'][$instance['bundle']]['properties'][$field['field_name']] = array();
       fivestar_property_info_callback($info, $entity_type, $field, $instance, 'fivestar');
       $entity_info = $info[$entity_type]['bundles'][$instance['bundle']]['properties'];
-
-      foreach ($entity_info[$field['field_name']]['property info'] as $property_name => $property) {
-        $microdata[$property_name]['#attributes'] = array();
-      }
+    }
+    foreach ($entity_info[$field['field_name']]['property info'] as $property_name => $property) {
+      $microdata[$property_name]['#attributes'] = array();
     }
   }
 
-- 
1.7.5.4

