diff --git a/includes/media.pages.inc b/includes/media.pages.inc
index 6414eb6..ebf2c8a 100644
--- a/includes/media.pages.inc
+++ b/includes/media.pages.inc
@@ -226,7 +226,7 @@ function media_add_upload_submit($form, &$form_state) {
   $file->display = TRUE;
 
   // Change the file status from temporary to permanent.
-  _media_save_file_permenently($file);
+  _media_save_file_permanently($file);
 
   // Determine what URI scheme this file should use.
   $scheme = !empty($params['uri_scheme']) && file_stream_wrapper_valid_scheme($params['uri_scheme']) ? $params['uri_scheme'] : file_default_scheme();
@@ -291,7 +291,7 @@ function media_add_upload_multiple_submit($form, &$form_state) {
 
       $file = file_uri_to_object($destination);
       file_save($file);
-      _media_save_file_permenently($file);
+      _media_save_file_permanently($file);
 
       $saved_files[] = $file;
     }
diff --git a/media.module b/media.module
index 63ff02f..3a12109 100644
--- a/media.module
+++ b/media.module
@@ -1038,7 +1038,7 @@ function media_filter_info() {
  * @param $file
  *  A file object.
  */
-function _media_save_file_permenently(&$file) {
+function _media_save_file_permanently($file) {
   if ($file->status < FILE_STATUS_PERMANENT) {
     $file->status = FILE_STATUS_PERMANENT;
     file_save($file);
