diff --git a/form_builder.module b/form_builder.module
index 195667b..84b8fec 100644
--- a/form_builder.module
+++ b/form_builder.module
@@ -354,5 +354,6 @@ function form_builder_registry_files_alter(&$files, $module_cache) {
   $module_files = file_scan_directory(drupal_get_path('module', 'form_builder') . '/modules', '!^.*\.inc$!');
   foreach ($module_files as $path => $file) {
     $files[$file->uri]['module'] = $file->name;
+    $files[$file->uri]['weight'] = 0;
   }
 }
