diff --git a/rotating_banner.module b/rotating_banner.module
index 48c4f1a..b434906 100644
--- a/rotating_banner.module
+++ b/rotating_banner.module
@@ -180,6 +180,7 @@ function rotating_banner_block_info() {
     foreach ($result as $row) {
       $blocks[$row->rbid] = array(
         'info' => t('Rotating banner: @title', array('@title' => $row->title)),
+        'cache' => DRUPAL_NO_CACHE,
       );
     }
   }
