Index: htmlpurifier.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/htmlpurifier/htmlpurifier.module,v
retrieving revision 1.21.4.3
diff -u -p -r1.21.4.3 htmlpurifier.module
--- htmlpurifier.module	12 Nov 2010 14:02:05 -0000	1.21.4.3
+++ htmlpurifier.module	9 Dec 2010 07:48:57 -0000
@@ -312,7 +312,10 @@ function _htmlpurifier_load() {
   $factory->register('Drupal', 'HTMLPurifier_DefinitionCache_Drupal');
 
   // Register the version as a variable:
-  variable_set('htmlpurifier_version_ours', HTMLPurifier::VERSION);
+  $current_version = variable_get('htmlpurifier_version_ours', FALSE);
+  if ($current_version != HTMLPurifier::VERSION) {
+    variable_set('htmlpurifier_version_ours', HTMLPurifier::VERSION);
+  }
 }
 
 /**
