diff --git a/fivestar.install b/fivestar.install
index 62e743b..2538ead 100644
--- a/fivestar.install
+++ b/fivestar.install
@@ -190,13 +190,15 @@ function fivestar_update_7204() {
           // Convert the formatter and set the exposed settings.
           $instance['display'][$key]['type'] == 'fivestar_formatter_default';
           $instance['display'][$key]['settings']['expose'] = TRUE;
-
+          $display = $instance['display'][$key];
+        }
+        if ($display['type'] == 'fivestar_formatter_default' && !empty($display['settings']['expose'])) {
           // The widget type needs to be exposed for the widget to be exposed.
           $instance['widget']['type'] = 'exposed';
 
           // If one of the displays allowed clearing change the field settings
           // to allow clearing.
-          if ($display['settings']['allow_clear'] == TRUE) {
+          if (!empty($display['settings']['allow_clear'])) {
             $instance['settings']['allow_clear'] = TRUE;
           }
         }
