diff --git a/file_entity/file_entity.file_api.inc b/file_entity/file_entity.file_api.inc
index 59e7588..2f04294 100644
--- a/file_entity/file_entity.file_api.inc
+++ b/file_entity/file_entity.file_api.inc
@@ -243,6 +243,7 @@ function file_view_file($file, $displays = 'default', $langcode = NULL) {
   // If none of the configured formatters were able to display the file, attempt
   // to display the file using the file type's default view callback.
   if (!isset($element)) {
+    $display = NULL;
     $file_type_info = file_info_file_types($file->type);
     if (isset($file_type_info['default view callback']) && ($function = $file_type_info['default view callback']) && function_exists($function)) {
       $element = $function($file, $view_mode, $langcode);
@@ -255,8 +256,10 @@ function file_view_file($file, $displays = 'default', $langcode = NULL) {
     $element += array(
       '#file' => $file,
       '#view_mode' => $view_mode,
+      '#display' => $display,
       '#language' => $langcode,
     );
+    drupal_alter('file_view_file', $element);
     return $element;
   }
 }
