diff --git a/media_gallery.install b/media_gallery.install
index dce80b4..dee3128 100644
--- a/media_gallery.install
+++ b/media_gallery.install
@@ -293,18 +293,15 @@ function _media_gallery_controlled_fields() {
       'field_name' => 'media_description',
       'locked' => TRUE,
       'type' => 'text_long',
-      'translatable' => TRUE,
     ),
     'media_title' => array(
       'field_name' => 'media_title',
       'locked' => TRUE,
       'type' => 'text',
-      'translatable' => TRUE,
     ),
     'field_license' => array(
       'field_name' => 'field_license',
       'locked' => TRUE,
-      'translatable' => TRUE,
       'settings' => array(
         'allowed_values_function' => '_media_gallery_get_field_license_values',
       ),
