diff --git a/server_variables.install b/server_variables.install
index 5765119..11f78c1 100644
--- a/server_variables.install
+++ b/server_variables.install
@@ -8,4 +8,14 @@ function server_variables_uninstall() {
   db_delete('variable')
     ->condition('name', 'server_variables_%', 'LIKE')
     ->execute();
+}
+
+/**
+ * Increase the module weight so that it will run after Metatag.
+ */
+function server_variables_update_7000() {
+  db_update('system')
+    ->fields(array('weight' => 2))
+    ->condition('name', 'server_variables', '=')
+    ->execute();
 }
\ No newline at end of file
diff --git a/server_variables.module b/server_variables.module
index b382fdb..f868376 100644
--- a/server_variables.module
+++ b/server_variables.module
@@ -213,6 +213,18 @@ function _server_variables_field_access_allowed($field, $entity_type, $entity) {
   return field_access('view', $field, $entity_type, $entity) || field_access('edit', $field, $entity_type, $entity);
 }
 
+
+/**
+ * Implements hook_preprocess_html().
+ */
+function server_variables_preprocess_html(&$variables) {
+  $vars = array();
+  if (_server_variables_check('site', 'page_title')) {
+    _server_variables_send($vars, 'site', 'page_title', $variables['head_title']);
+  }
+  drupal_add_js(array('server_variables' => $vars), 'setting');
+}
+
 /**
  * Implements hook_init().
  *  - set Drupal.settings from session object
@@ -225,9 +237,6 @@ function server_variables_init() {
     if (_server_variables_check('site', 'site_name')) {
       _server_variables_send($vars, 'site', 'site_name', variable_get('site_name', ''));
     }
-    if (_server_variables_check('site', 'page_title')) {
-      _server_variables_send($vars, 'site', 'page_title', drupal_get_title());
-    }
     if (user_is_logged_in()) {
       // be sure user is logged in
       // get variables for currently logged in user
@@ -298,7 +307,7 @@ function server_variables_entity_delete($entity, $type) {
  * @param $op the operation (eg. "view", "insert", "update", "delete")
  */
 function _server_variables_entity_send($entity, $type, $op) {
-  if (($type == 'user' && $entity->uid == 1) || ($type == 'redirect')) {
+  if (($type == 'user' && $entity->uid == 1) || ($type == 'redirect') || ($type == 'file')) {
     // Never run for user 1 or for redirect entities.
     return;
   }
@@ -360,14 +369,14 @@ function server_variables_library() {
   $path = drupal_get_path('module', 'server_variables');
   // Register the debug jQuery plugin as a library.
   $libraries['server_variables.debug'] = array(
-    'title' => 'Server variables debug tool',
+    'title' => 'Analytics debug tool',
     'website' => 'http://drupal.org/project/server_variables',
     'version' => '1.0',
     'js' => array(
       $path . '/tests/jquery.client-variable-debug.js' => array('group' => JS_THEME, 'preprocess' => false),
     ),
   );
-  
+
   return $libraries;
 }
 
