diff --git a/node_export.module b/node_export.module
index 8bdadb9..33305a2 100755
--- a/node_export.module
+++ b/node_export.module
@@ -964,6 +964,7 @@ function node_export_format_handlers() {
   if (empty($format_handlers)) {
     $format_handlers = module_invoke_all('node_export_format_handlers');
   }
+  drupal_alter('node_export_format_handlers', $format_handlers);
   return $format_handlers;
 }
 
