diff --git a/modules/webform/form_builder_webform.components.inc b/modules/webform/form_builder_webform.components.inc
index b9ad09b..770738a 100644
--- a/modules/webform/form_builder_webform.components.inc
+++ b/modules/webform/form_builder_webform.components.inc
@@ -32,7 +32,7 @@ function _form_builder_webform_form_builder_types_email() {
       '#type' => 'textfield',
       '#theme_wrappers' => array('webform_element'),
       '#form_builder' => array('element_type' => 'email'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
@@ -126,7 +126,7 @@ function _form_builder_webform_form_builder_types_file() {
       '#title' => t('New file'),
       '#type' => 'file',
       '#theme_wrappers' => array('webform_element'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
       '#webform_file_filtering' => $defaults['extra']['filtering'],
       '#webform_file_savelocation' => $defaults['extra']['savelocation'],
       '#webform_file_width' => $defaults['extra']['width'],
@@ -396,7 +396,7 @@ function _form_builder_webform_form_builder_types_markup() {
       '#theme_wrappers' => array('webform_element'),
       '#markup' => $default,
       '#format' => NULL,
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
@@ -535,7 +535,7 @@ function _form_builder_webform_form_builder_types_select() {
       '#options' => array('one' => 'one', 'two' => 'two', 'three' => 'three'),
       '#default_value' => 'one',
       '#webform' => array('type' => 'select'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
@@ -559,7 +559,7 @@ function _form_builder_webform_form_builder_types_select() {
       '#default_value' => array('one', 'two'),
       '#multiple' => TRUE,
       '#webform' => array('type' => 'select'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
@@ -581,7 +581,7 @@ function _form_builder_webform_form_builder_types_select() {
       '#options' => array('one' => 'one', 'two' => 'two', 'three' => 'three'),
       '#default_value' => 'one',
       '#webform' => array('type' => 'select'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
@@ -650,7 +650,7 @@ function _form_builder_webform_form_builder_types_textarea() {
       '#title' => t('New textarea'),
       '#type' => 'textarea',
       '#theme_wrappers' => array('webform_element'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
@@ -701,7 +701,7 @@ function _form_builder_webform_form_builder_types_textfield() {
       '#title' => t('New textfield'),
       '#type' => 'textfield',
       '#theme_wrappers' => array('webform_element'),
-      '#title_display' => 'above',
+      '#title_display' => 'before',
     ),
   );
 
diff --git a/modules/webform/form_builder_webform.module b/modules/webform/form_builder_webform.module
index b6348e4..e63cd6d 100644
--- a/modules/webform/form_builder_webform.module
+++ b/modules/webform/form_builder_webform.module
@@ -415,11 +415,9 @@ function form_builder_webform_form_form_builder_field_configure_alter(&$form, &$
   // Change the list of options from the default Drupal values to the acceptable
   // webform values.
   $form['title_display']['#options'] = array(
-    'above' => 'Above',
-    'below' => 'Below',
-    'before' => 'Inline before',
-    'after' => 'Inline after',
-    'invisible' => 'Hidden',
+    'before' => t('Above'),
+    'inline' => t('Inline'),
+    'none' => t('Hidden'),
   );
-  $form['title_display']['#default_value'] = isset($form['title_display']['#default_value']) ? $form['title_display']['#default_value'] : 'above';
+  $form['title_display']['#default_value'] = isset($form['title_display']['#default_value']) ? $form['title_display']['#default_value'] : 'before';
 }
