diff --git a/formats/xml.inc b/formats/xml.inc
index 27978f5..cf8263f 100644
--- a/formats/xml.inc
+++ b/formats/xml.inc
@@ -189,6 +189,9 @@ class NodeExportXmlDecoder {
           return NULL;
         }
       }
+      elseif (!empty($stack['attributes']['_NUMERIC_KEYS'])) {
+        return array();
+      }
       return htmlspecialchars_decode(trim($stack['data']));
     }
     else {
@@ -229,4 +232,4 @@ class NodeExportXmlDecoder {
  */
 function node_export_xml_action_form($context, &$form_state) {
   return node_export_action_form($context, $form_state, 'xml');
-}
\ No newline at end of file
+}
