diff --git a/views.module b/views.module
index 4bc8871..0f6f0a3 100644
--- a/views.module
+++ b/views.module
@@ -1312,7 +1312,7 @@ function views_get_applicable_views($type) {
   // @todo: Use a smarter flagging system so that we don't have to
   // load every view for this.
   $result = array();
-  $views = views_get_all_views();
+  $views = views_get_all_views(TRUE);
 
   foreach ($views as $view) {
     // Skip disabled views.
