diff --git a/metatag.module b/metatag.module
index ce1514c..a12b4fc 100644
--- a/metatag.module
+++ b/metatag.module
@@ -730,10 +730,10 @@ function metatag_load_entity_from_path($path) {
  * Implements hook_page_build().
  */
 function metatag_page_build(&$page) {
-  // For some reason with Overlay enabled we get an empty $page, so just fail
-  // this case.
+  // If there is no main page content yet, initialize it as an array so we can
+  // add to it below.
   if (!isset($page['content'])) {
-    return;
+    $page['content'] = array();
   }
 
   // Load the metatags render array in before any page content so that more
