--- a/plugins/views_plugin_display_page.inc
+++ b/plugins/views_plugin_display_page.inc
@@ -474,6 +474,9 @@ class views_plugin_display_page extends views_plug
         // automatically remove '/' from path.
         $form_state['values']['path'] = trim($form_state['values']['path'], '/');
 
+        // remove trailing spaces
+        $form_state['values']['path'] = trim($form_state['values']['path']);
+
         break;
       case 'menu':
         $path = $this->get_option('path');
