Index: htmlpurifier.admin.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/htmlpurifier/htmlpurifier.admin.inc,v
retrieving revision 1.1.2.1
diff -u -p -r1.1.2.1 htmlpurifier.admin.inc
--- htmlpurifier.admin.inc	27 Oct 2010 17:13:19 -0000	1.1.2.1
+++ htmlpurifier.admin.inc	9 Dec 2010 07:45:13 -0000
@@ -22,13 +22,6 @@ function htmlpurifier_admin_settings($fo
     '#submit' => array('_htmlpurifier_clear_cache')
   );
 
-  $form['htmlpurifier_doublecache'] = array(
-    '#type' => 'checkbox',
-    '#title' => t('Allow double caching'),
-    '#default_value' => FALSE,
-    '#description' => t('If enabled, HTML Purifier will tell the Filter module that its output is cacheable. This is not usually necessary, because HTML Purifier maintains its own cache, but may be helpful if you have later filters that need to be cached. Note: This applies to all text formats in which the HTML Purifier filters are included.'),
-  );
-
   $form = system_settings_form($form);
   $form['#submit'][] = 'htmlpurifier_admin_settings_submit';
   return $form;
Index: htmlpurifier.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/htmlpurifier/htmlpurifier.module,v
retrieving revision 1.21.4.3
diff -u -p -r1.21.4.3 htmlpurifier.module
--- htmlpurifier.module	12 Nov 2010 14:02:05 -0000	1.21.4.3
+++ htmlpurifier.module	9 Dec 2010 07:45:13 -0000
@@ -99,9 +99,6 @@ function htmlpurifier_filter_info() {
       'htmlpurifier_help' => TRUE,
     ),
     'tips callback' => '_htmlpurifier_filter_tips',
-    // Since HTML Purifier implements its own caching layer, having filter
-    // cache it again is wasteful. Returns TRUE if double caching is permitted.
-    'cache' => variable_get("htmlpurifier_doublecache", FALSE),
   );
   $filters['htmlpurifier_advanced'] = array(
     'title' => t('HTML Purifier (advanced)'),
