diff --git a/extlink.info b/extlink.info
index a90b061..7fc8ce9 100644
--- a/extlink.info
+++ b/extlink.info
@@ -1,6 +1,6 @@
 ; $Id: extlink.info,v 1.4 2010/10/17 00:42:23 quicksketch Exp $
 name = External Links
-description = Adds icon next to external links on the site.
+description = Determine behavior and appearance for external links. Options include: Make external links open in a new window, add icons next to external links and email addresses, display pop-up warnings when users click external links.
 package = User interface
 core = 7.x
 configure = admin/config/user-interface/extlink
diff --git a/extlink.module b/extlink.module
index 70d6f56..b4ad71c 100644
--- a/extlink.module
+++ b/extlink.module
@@ -40,31 +40,23 @@ function extlink_admin_settings() {
 
   $form['extlink_class'] = array(
     '#type' => 'checkbox',
-    '#title' => t('Add icon to external links'),
+    '#title' => t('Place an icon next to external links.'),
     '#return_value' => 'ext',
     '#default_value' => variable_get('extlink_class', 'ext'),
-    '#description' => t('Places an !icon icon next to external links.', array('!icon' => theme_image(array('path' => drupal_get_path('module', 'extlink') .'/extlink.png', 'alt' => t('External Links icon'), 'attributes' => array())))),
   );
 
   $form['extlink_mailto_class'] = array(
     '#type' => 'checkbox',
-    '#title' => t('Add icon to mailto links'),
+    '#title' => t('Place an icon next to mailto links'),
     '#return_value' => 'mailto',
     '#default_value' => variable_get('extlink_mailto_class', 'mailto'),
-    '#description' => t('Places an !icon icon next to mailto links.', array(
-      '!icon' => theme_image(array(
-        'path' => drupal_get_path('module', 'extlink') . '/mailto.png',
-        'alt' => t('Email links icon'),
-        'attributes' => '',
-      )),
-    )),
   );
 
   $form['extlink_subdomains'] = array(
     '#type' => 'checkbox',
-    '#title' => t('Consider subdomains internal'),
+    '#title' => t('Exclude links with the same primary domain.'),
     '#default_value' => variable_get('extlink_subdomains', 1),
-    '#description' => t('If checked, links with the same primary domain will all be considered internal. A link from www.example.com to my.example.com would be considered internal. Links between the www. and non-www. domain are always considered internal.'),
+    '#description' => t("For example, a link from 'www.example.com' to the subdomain of 'my.example.com' would be excluded."),
   );
 
   $form['extlink_target'] = array(
@@ -72,24 +64,27 @@ function extlink_admin_settings() {
     '#title' => t('Open external links in a new window'),
     '#return_value' => '_blank',
     '#default_value' => variable_get('extlink_target', 0),
-    '#description' => t('Should all external links be opened in a new window?'),
   );
 
   $form['extlink_alert'] = array(
     '#type' => 'checkbox',
-    '#title' => t('Display pop-up warnings'),
+    '#title' => t('Display a pop-up warning when any external link is clicked.'),
     '#return_value' => '_blank',
     '#default_value' => variable_get('extlink_alert', 0),
-    '#description' => t('Displays a pop-up warning when any external link is clicked.'),
   );
 
   $form['extlink_alert_text'] = array(
     '#type' => 'textarea',
+    '#title' => t('Text to display in the pop-up warning box.'),
     '#rows' => 3,
-    '#title' => t('Pop-up warning text'),
-    '#default_value' => variable_get('extlink_alert_text', 'This link will take you to an external web site. We are not responsible for their content.'),
-    '#description' => t('Text to display in the pop-up external link warning box.'),
+    '#default_value' => variable_get('extlink_alert_text', 'This link will take you to an external web site.'),
     '#wysiwyg' => FALSE,
+    '#states' => array(
+      // Only show this field when user opts to display a pop-up warning.
+      'visible' => array(
+        ':input[name="extlink_alert"]' => array('checked' => TRUE),
+      ),
+    ),
   );
 
   $patterns = array(
