diff --git a/includes/fivestar.field.inc b/includes/fivestar.field.inc
index c707141..2c3a6df 100644
--- a/includes/fivestar.field.inc
+++ b/includes/fivestar.field.inc
@@ -380,8 +380,11 @@ function fivestar_field_formatter_settings_form($field, $instance, $view_mode, $
     // Removed entity_get_info here and simplified the text. This is not saved
     // to used here and is honestly not need it. See http://drupal.org/node/1268972
     $element['expose'] = array(
-      '#type' => 'checkbox',
-      '#title' => t('Expose this Fivestar field for voting on the @type type.'),
+      '#type' => 'radios',
+      '#options' => array(
+        t('Disallow Fivestar voting on this display.'),
+        t('Allow Fivestar voting on this display.'),
+      ),
       '#default_value' => $settings['expose'],
       '#return_value' => 1
     );
