diff --git a/js/media.popups.js b/js/media.popups.js
index 9a615c5..a7bf374 100644
--- a/js/media.popups.js
+++ b/js/media.popups.js
@@ -101,8 +101,9 @@ Drupal.media.popups.mediaBrowser = function (onSelect, globalOptions, pluginOpti
 Drupal.media.popups.mediaBrowser.mediaBrowserOnLoad = function (e) {
   var options = e.data;
   if (this.contentWindow.Drupal.media.browser.selectedMedia.length > 0) {
-    var ok = $(this).dialog('option', 'buttons')['OK'];
-    ok.call(this);
+    var ok = (Drupal && Drupal.t) ? Drupal.t('OK') : 'OK';
+    var ok_func = $(this).dialog('option', 'buttons')[ok];
+    ok_func.call(this);
     return;
   }
 };
