diff --git a/css/fivestar.css b/css/fivestar.css
index 0dc1c2a..c467e1a 100644
--- a/css/fivestar.css
+++ b/css/fivestar.css
@@ -12,7 +12,13 @@ form.fivestar-widget input {
 .fivestar-combo-stars .fivestar-static-form-item {
   float: left; /* RTL */
   margin-right: 40px; /* RTL */
+  margin-left: 40px; /* RTL */
 }
+.fivestar-combo-stars .fivestar-static-item {
+  float: left; /* RTL */
+  margin-right: 40px; /* RTL */
+}
+
 .fivestar-combo-stars .fivestar-form-item {
   float: left; /* RTL */
 }
diff --git a/fivestar.module b/fivestar.module
index e6e0023..4a9df9e 100644
--- a/fivestar.module
+++ b/fivestar.module
@@ -466,14 +466,12 @@ function fivestar_expand($element) {
     '#required' => $element['#required'],
     '#theme' => 'fivestar_select',
     '#default_value' => _fivestar_get_element_default_value($element),
+    '#weight' => -2,
   );
 
   if (isset($element['#parents'])) {
     $element['vote']['#parents'] = $element['#parents'];
   }
-  if (isset($element['#weight'])) {
-    $element['vote']['#weight'] = $element['#weight'];
-  }
 
   switch ($settings['text']) {
     case 'user':
diff --git a/includes/fivestar.theme.inc b/includes/fivestar.theme.inc
index 4a74fdf..8038abc 100644
--- a/includes/fivestar.theme.inc
+++ b/includes/fivestar.theme.inc
@@ -35,7 +35,7 @@ function theme_fivestar_preview_widget($variables) {
       'css' => isset($variables['css']) && $variables['css'] != 'default' ? $variables['css'] : FALSE,
     ),
   );
-  
+
   // Attach necessary JS settings
   $settings = array(
     'titleUser' => t('Your rating') .': ',
@@ -90,6 +90,11 @@ function theme_fivestar_preview_wrapper($variables) {
 
 /**
  * Theme function for 'default' fivestar field formatter.
+ *
+ * This themes static stars. That is, pairs of stars where neither set
+ * of stars is "exposed". Exposed stars are clickable and displayed in a
+ * form. Theming of exposed stars is handled by the form array (and calls
+ * the same theme functions called at the end of this function).
  */
 function theme_fivestar_formatter_default($variables) {
   $element = $variables['element'];
@@ -108,7 +113,7 @@ function theme_fivestar_formatter_default($variables) {
     'widget' => $element['#widget'],
   );
   $star_display = theme('fivestar_static', $variables);
-  return theme('fivestar_static_element', array('description' => $element['#description'], 'star_display' => $star_display));
+  return theme('fivestar_static_element', array('description' => $element['#description'], 'star_display' => $star_display, 'is_form' => FALSE));
 }
 
 /**
@@ -124,7 +129,7 @@ function theme_fivestar_formatter_rating($variables) {
   $stars = (empty($element['#instance_settings']['stars'])) ? 5 : $element['#instance_settings']['stars'];
   $average = $element['#item']['average'];
   // Rating is X out of Y stars.
-  $rating = round(($average/100) * $stars, 1); 
+  $rating = round(($average/100) * $stars, 1);
   $output = $rating . '/' . $stars;
 
   return $output;
@@ -182,7 +187,7 @@ function theme_fivestar_static($variables) {
   if($widget['name'] != 'default') {
     drupal_add_css($widget['css']);
   }
-  
+
   $output = '<div class="fivestar-' . $widget['name'] . '">';
   $output .= '<div class="fivestar-widget-static fivestar-widget-static-'. $tag .' fivestar-widget-static-'. $stars . ' clearfix">';
   if (empty($stars)) {
@@ -281,7 +286,12 @@ function theme_fivestar_summary($variables) {
  */
 function theme_fivestar_static_element($variables) {
   $output = '';
-  $output .= '<div class="fivestar-static-form-item">';
+  if (isset($variables['is_form']) && !$variables['is_form']) {
+    $output .= '<div class="fivestar-static-item">';
+  }
+  else {
+    $output .= '<div class="fivestar-static-form-item">';
+  }
   $element = array(
     '#type' => 'item',
     '#title' => $variables['title'],
