diff --git a/form_builder.module b/form_builder.module
index 84b8fec..b339fb7 100644
--- a/form_builder.module
+++ b/form_builder.module
@@ -101,7 +101,7 @@ function form_builder_block_info() {
  */
 function form_builder_block_view($delta = '') {
   $block = array();
-  if ($delta == 'fields') {
+  if ($delta == 'fields' && form_builder_active_form()) {
     module_load_include('inc', 'form_builder', 'includes/form_builder.admin');
     $block['content'] = form_builder_field_palette();
   }
