diff --git a/css/views-admin-rtl.css b/css/views-admin-rtl.css
index cb3edaa..f7e4e7c 100644
--- a/css/views-admin-rtl.css
+++ b/css/views-admin-rtl.css
@@ -5,6 +5,10 @@
 
 /* @group Inline lists */
 
+.horizontal-container {
+  float: right;
+}
+
 .horizontal > * {
   float: right;
 }
diff --git a/css/views-admin.css b/css/views-admin.css
index 2582900..711e77c 100644
--- a/css/views-admin.css
+++ b/css/views-admin.css
@@ -25,6 +25,10 @@
 
 /* @group Inline lists */
 
+.horizontal-container {
+  float: left; /* LTR */
+}
+
 .horizontal > * {
   clear: none;
   float: left; /* LTR */
diff --git a/modules/field/views_handler_field_field.inc b/modules/field/views_handler_field_field.inc
index 05f1678..61c89f0 100644
--- a/modules/field/views_handler_field_field.inc
+++ b/modules/field/views_handler_field_field.inc
@@ -361,6 +361,10 @@ class views_handler_field_field extends views_handler_field {
       ),
       '#submit' => array('views_ui_config_item_form_submit_temporary'),
       '#executes_submit_callback' => TRUE,
+      // Display this horizontally along with the associated settings form (see
+      // $form['settings'] below).
+      '#prefix' => '<div class="horizontal-container">',
+      '#suffix' => '</div>',
     );
 
     $form['field_api_classes'] = array(
@@ -399,6 +403,14 @@ class views_handler_field_field extends views_handler_field {
       $settings_form = $function($field, $instance, '_dummy', $form, $form_state);
     }
     $form['settings'] = $settings_form;
+    // Display this horizontally along with the associated settings form (see
+    // $form['type'] above). Avoid clobbering any existing prefix or suffix.
+    $form['settings'] += array(
+      '#prefix' => '',
+      '#suffix' => '',
+    );
+    $form['settings']['#prefix'] = '<div class="dependent-options horizontal-container clearfix">' . $form['settings']['#prefix'];
+    $form['settings']['#suffix'] .= '</div>';
   }
 
   /**
