diff --git a/modules/comment.views.inc b/modules/comment.views.inc
index e22ccd5..f3315a5 100644
--- a/modules/comment.views.inc
+++ b/modules/comment.views.inc
@@ -505,7 +505,7 @@ function comment_views_data() {
       'handler' => 'views_handler_argument_numeric',
     ),
     'field' => array(
-      'handler' => 'views_handler_filter_user_name',
+      'handler' => 'views_handler_field_user',
     ),
   );
 
