diff --git a/pathauto.test b/pathauto.test
index 4938e21..3ec6fdd 100644
--- a/pathauto.test
+++ b/pathauto.test
@@ -318,18 +318,18 @@ class PathautoUnitTestCase extends PathautoTestHelper {
     $this->assertNoEntityAlias('node', $node);
 
     // Check that Pathauto does not create an alias of '/modules'.
-    $node->title = 'Modules';
+    $node = $this->drupalCreateNode(array('title' => 'Modules', 'type' => 'page'));
     node_save($node);
     $this->assertNoEntityAlias('node', $node);
 
     // Check that Pathauto does not create an alias of '/index.php'.
-    $node->title = 'index.php';
+    $node = $this->drupalCreateNode(array('title' => 'index.php', 'type' => 'page'));
     node_save($node);
     $this->assertNoEntityAlias('node', $node);
 
     // Check that a safe value gets an automatic alias. This is also a control
     // to ensure the above tests work properly.
-    $node->title = 'Safe value';
+    $node = $this->drupalCreateNode(array('title' => 'Safe value', 'type' => 'page'));
     node_save($node);
     $this->assertEntityAlias('node', $node, 'safe-value');
   }
