diff --git a/rotating_banner.module b/rotating_banner.module
index ef21b4d..48c4f1a 100644
--- a/rotating_banner.module
+++ b/rotating_banner.module
@@ -238,8 +238,14 @@ function rotating_banner_block_configure($delta) {
   $rb_form['slides']['slide_table']['weight'] = array('#tree' => TRUE);
   
   foreach ($slides as $slide) {
-    $media = media_load($slide->fid);
-    $preview = field_view_field('media', $media, 'file', 'media_preview');
+    if (module_exists('file_entity')) {
+      $loaded_file = file_load($slide->fid);
+      $preview = file_view_file($loaded_file, 'media_preview');
+    }
+    else {
+      $media = media_load($slide->fid);
+      $preview = field_view_field('media', $media, 'file', 'media_preview');
+    }
     $preview['#theme_wrappers'][] = 'media_thumbnail';
 
     $i++;
